autofs: enable NIS support

BarrOff opened the pull request:
The current autofs package does not support automounter maps from [NIS](https://en.wikipedia.org/wiki/Network_Information_Service).
The configure script of autofs searches for the header file `rpcsvc/ypclnt.h` file and enables NIS support when it finds it.
Adding the `libnsl.dev` package to `buildInputs` makes the header available during the build process.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Tested against NFS shares from a real NIS server

FliegendeWurst commented on pkgs/os-specific/linux/autofs/default.nix:
```diff
@@ -48,9 +48,9 @@ stdenv.mkDerivation rec {
+                  openssl cyrus_sasl rpcsvc-proto libnsl ];
 
-  nativeBuildInputs = [ flex bison pkgconf ];
+  nativeBuildInputs = [ flex bison pkgconf libnsl.dev ];
```
```suggestion
  nativeBuildInputs = [ flex bison pkgconf ];
```
Seems to build with NIS support without this too.

BarrOff replied:
I just tried it without this and autofs was not able to lookup mounts through NIS, so I think it is indeed required

BarrOff committed:
autofs5: enable NIS support

The current package does not support reading NIS maps of autofs
templates. Adding the libnsl.dev package to buildInputs lets the
configure script detect rpcsvc/ypclnt.h which enables support for NIS.

BarrOff commented:
@wegank

I rebased the change on top of the `pkgs/by-name` migration so there is no merge conflict any more.

FliegendeWurst commented:
By the way, there is no need to tag wegank. The labels are set automatically.
